src/Sharpliner.Model/GHActions/Webhook.cs

```diff
@@ -0,0 +1,450 @@
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace Sharpliner.Model.GHActions
+{
+    /// <summary>
+    /// Base abstract class for all webhook related triggers.
+    /// </summary>
+    public abstract record Webhook : Event;
+
+    // All the following enum types look verbose.. but we are
+    // looking into the future with https://github.com/dotnet/csharplang/issues/2926
+    // ideally, all these will look as nice as:
+    //
+    // var cr = new CheckRun {
+    //     Activities = { Created, Requested }
+    // }
+    //
+    // we will have to wait until c# does implement the feature, so for the time
+    // being we will do:
+    //
+    // var cr = new CheckRun {
+    //     Activities = { CheckRun.Activity.Created, CheckRun.Activity.Requested }
+    // }
+    //
+    // verbose, but type safe and autocomplete will help. If not, get a better editor!
```

Comment:
@premun this will make  it very nice, they are working on it.

Reply:
Nice, also didn't know this is coming!

Reply:
We could potentially live on top of net6.0 and have the `LangVersion` set to `latest` and get this early :)